stablehlo/tests/stablehlo_canonicalize_dynamism.mlir

```diff
@@ -1,7 +1,97 @@
-// RUN: stablehlo-opt --stablehlo-canonicalize-dynamism %s | FileCheck %s
+// RUN: stablehlo-opt --stablehlo-canonicalize-dynamism --split-input-file --verify-diagnostics %s | FileCheck %s
 
-// CHECK-LABEL: func @canonicalize_dynamic_conv
-func.func @canonicalize_dynamic_conv(%arg0: tensor<100x26x26x32xf32>, %arg1: tensor<3x3x1x32xf32>) -> tensor<100x28x28x1xf32> {
+// CHECK-LABEL: func @custom_call_success
+func.func @custom_call_success(%arg0: tensor<4xf32>) -> (tensor<1x2xf32>, tensor<3x4xf32>) {
```

Comment:
Would be good to see a few more positive tests to document what's ok under this API. A few thoughts for tests, if the behavior shouldn't be allowed, consider adding verifiers to `Base.cpp` for them:

- Repeating `indices_of_shape_operands = [1, 1]`, given that we had the recent bug in refinement with repeated return arguments.
- Indices of shape argument not last `custom_call @foo(%0, %1) {
    indices_of_shape_operands = dense<[0]> : tensor<2xi64>
  } : (tensor<2xi64>, tensor<4xf32>) -> (tensor<3x4xf32>)`
- Constant operand used in `indices_of_shape_operands` has negative value `[1, -1]`

Reply:
Both this comment and the comment above are good ideas. Will address them in a child PR which refactors the tests for this pass.